packages/wrangler/src/delete.ts

```diff
@@ -71,10 +71,27 @@ export async function deleteHandler(args: DeleteArgs) {
 	);
 
 	if (confirmed) {
+		let force: boolean;
+
+		const extraConfirmed = await checkAndConfirmForceDeleteIfNecessary(
+			scriptName,
+			accountId
+		);
+		switch (extraConfirmed) {
+			case "not_needed":
+				force = false;
+				break;
+			case "needed_confirmed":
+				force = true;
+				break;
+			case "needed_denied":
+				return;
+		}
+
 		await fetchResult(
 			`/accounts/${accountId}/workers/services/${scriptName}`,
 			{ method: "DELETE" },
-			new URLSearchParams({ force: "true" })
+			new URLSearchParams({ force: force.toString() })
```

Comment:
What kind of error response does the API return in the `"needed_denied"` case?

Reply:
Depends on what you mean - in the "needed_denied" case, we don't make any call to the API. The user has rejected the confirmation for using extra aggressive force-deletes, so we return early.
But if you're asking what would happen if we tried to delete with `?force=false` when the Worker is a dependent service, then the API would return a Forbidden error with a message describing which resource is blocking the delete.